### ISSUE TITLE
AtomCertVerifier should be initialized in IO thread

### ISSUE THREAD

**Issue by qrohlf** [electron/electron#6261]
* Electron version: 1.2.3
* Operating system: Ubuntu server 12.04

I'm getting the following error trying to start electron-prebuilt on my CI box:

```
Received signal 6
#0 0x000001d95c3b <unknown>
#1 0x7fbcff6c5cb0 <unknown>
#2 0x7fbcf929b0d5 gsignal
#3 0x7fbcf929e83b abort
#4 0x000001d948e2 <unknown>
#5 0x000001dac09a <unknown>
#6 0x000001b72c5a <unknown>
#7 0x00000075359b <unknown>
#8 0x0000007dbd3e <unknown>
#9 0x000000754984 <unknown>
#10 0x000003369775 atom::AtomCertVerifier::AtomCertVerifier()
#11 0x0000034026aa atom::AtomBrowserContext::AtomBrowserContext()
#12 0x0000034040bb brightray::BrowserContext::Create()
#13 0x0000033d5302 brightray::BrowserContext::From()
#14 0x0000033d742f brightray::BrowserMainParts::PreMainMessageLoopRun()
#15 0x0000032cca12 atom::AtomBrowserMainParts::PreMainMessageLoopRun()
#16 0x00000259a16a <unknown>
#17 0x000002887457 <unknown>
#18 0x000002598520 <unknown>
#19 0x00000259ce74 <unknown>
#20 0x0000025958f6 <unknown>
#21 0x0000011b8c97 <unknown>
#22 0x0000011b76b0 <unknown>
#23 0x0000032c9650 main
#24 0x7fbcf928676d __libc_start_main
#25 0x000000560df9 <unknown>
  r8: 00007fffe132ebb0  r9: 00007fbcf93a6557 r10: 0000000000000008 r11: 0000000000000202
 r12: 0000108e15274e00 r13: 000000000259caf0 r14: 00007fffe132ef38 r15: 00007fffe132ef28
  di: 000000000000790b  si: 000000000000790b  bp: 0000000000000000  bx: 0000000000000000
  dx: 0000000000000006  ax: 0000000000000000  cx: ffffffffffffffff  sp: 00007fffe132e658
  ip: 00007fbcf929b0d5 efl: 0000000000000202 cgf: 0000000000000033 erf: 0000000000000007
 trp: 000000000000000e msk: 0000000000000000 cr2: 00007fe3fbe31680
[end of stack trace]
```

Any ideas for how to debug? At a wild guess, maybe I'm missing some SSL related library headers since there's that `AtomCertVerifier` in the stack trace, but some kind of error message would be a lot more helpful than a silent crash and a stack dump.

**Comment by zcbenz** [comment]
Are you running your own app or just starting the default app?

**Comment by qrohlf** [comment]
@zcbenz just starting the default app

**Comment by zcbenz** [comment]
It is probably because `AtomCertVerifier` is being created in UI thread while we should init it in IO thread. I don't know why no one saw this crash before.

**Comment by qrohlf** [comment]
@zcbenz has this crash been fixed in 1.2.6? If not, would this make a decent first bug for me to hack on in this project? The current issue is preventing my team from being able to set up CI for our project that uses electron.
